site/en/docs/extensions/mv3/nativeMessaging/index.md

```diff
@@ -0,0 +1,202 @@
+registry.
+
+On **macOS** and **Linux**, the location of the native messaging host's manifest file varies by the
+browser (Google Chrome or Chromium). The system-wide native messaging hosts are looked up at a fixed
+location, while the user-level native messaging hosts are looked up in a subdirectory within the
+[user profile directory][user-data-directory] called `NativeMessagingHosts`.
+
+- macOS (system-wide)
+  - Google Chrome: `/Library/Google/Chrome/NativeMessagingHosts/com.my_company.my_application.json`
+  - Chromium: `/Library/Application Support/Chromium/NativeMessagingHosts/com.my_company.my_application.json`
+- macOS (user-specific, _default_ path)
+  - Google Chrome: `~/Library/Application Support/Google/Chrome/NativeMessagingHosts/com.my_company.my_application.json`
+  - Chromium: `~/Library/Application Support/Chromium/NativeMessagingHosts/com.my_company.my_application.json`
+- Linux (system-wide)
+  - Google Chrome: `/etc/opt/chrome/native-messaging-hosts/com.my_company.my_application.json`
+  - Chromium: `/etc/chromium/native-messaging-hosts/com.my_company.my_application.json`
+- Linux (user-specific, _default_ path)
+  - Google Chrome: `~/.config/google-chrome/NativeMessagingHosts/com.my_company.my_application.json`
+  - Chromium: `~/.config/chromium/NativeMessagingHosts/com.my_company.my_application.json`
+
+### Native messaging protocol {: #native-messaging-host-protocol }
+
+Chrome starts each native messaging host in a separate process and communicates with it using
+standard input (`stdin`) and standard output (`stdout`). The same format is used to send messages in
+both directions: each message is serialized using JSON, UTF-8 encoded and is preceded with 32-bit
+message length in native byte order. The maximum size of a single message from the native messaging
+host is 1 MB, mainly to protect Chrome from misbehaving native applications. The maximum size of the
+message sent to the native messaging host is 4 GB.
+
+The first argument to the native messaging host is the origin of the caller, usually
+`chrome-extension://[ID of allowed extension]`. This allows native messaging hosts to identify the
+source of the message when multiple extensions are specified in the `allowed_origins` key in the
+[native messaging host manifest][native-messaging-host-manifest].
+**_Warning_**: In Windows, in Chrome 54 and earlier, the origin was passed as the second parameter
+instead of the first parameter.
+
+When a messaging port is created using [runtime.connectNative][connect-native] Chrome starts native messaging
+host process and keeps it running until the port is destroyed. On the other hand, when a message is
+sent using [runtime.sendNativeMessage][send-native-message], without creating a messaging port, Chrome starts a new
+native messaging host process for each message. In that case the first message generated by the host
+process is handled as a response to the original request, i.e. Chrome will pass it to the response
```

Comment:
This was a direct explanation of "handled as a response", which is an ambiguous concept without its own specific meaning, so it might be better to leave the explanation in the same sentence. In case you're sure starting a sentence reads better consider starting it with `That is` instead of a somewhat unusually placed `I.e.`.

Reply:
I've reworded to just avoid this entirely.